PhysicsTools/PatUtils/interface/ShiftedJetProducerT.h

```diff
@@ -108,8 +108,9 @@ template <typename T, typename Textractor>
     edm::Handle<reco::JetCorrector> jetCorrUpToL3;
     evt.getByToken(jetCorrTokenUpToL3_, jetCorrUpToL3);
     edm::Handle<reco::JetCorrector> jetCorrUpToL3Res;
-    evt.getByToken(jetCorrTokenUpToL3Res_, jetCorrUpToL3Res);
-
+    if ( evt.isRealData() && addResidualJES_ ) {
+      evt.getByToken(jetCorrTokenUpToL3Res_, jetCorrUpToL3Res);
```

Comment:
This is already the, but the conditional value was on the jetCorrTokenUpToL3Res_ value itself. I added the addResidualJES_ condition as well.